types/components/Button.d.ts

```diff
@@ -0,0 +1,34 @@
+import { BoolProps, Breakpoint, GeneratedPropTypes } from '../types';
+import React from 'react';
+export declare const buttonColorVariations: {
+    default: import('styled-components').RuleSet<object>;
+    error: import('styled-components').RuleSet<object>;
+    gray: import('styled-components').RuleSet<object>;
+    info: import('styled-components').RuleSet<object>;
+    secondary: import('styled-components').RuleSet<object>;
+    success: import('styled-components').RuleSet<object>;
+};
+export declare const buttonSizeVariations: {
+    default: import('styled-components').RuleSet<object>;
+    large: import('styled-components').RuleSet<object>;
+    xl: import('styled-components').RuleSet<object>;
+};
+declare type ButtonBaseProps = {
+    as?: any;
+    className?: string;
+    children?: any | any[];
+    disabled?: boolean;
+    href?: string;
+    icon?: string;
+    fluid?: Breakpoint | boolean;
+    onClick?: any;
```

Comment:
can this be something like
```suggestion
    onClick?: () => void;
```
?